### ISSUE TITLE
`<button>` inside `<select>` inside `<label>` throw error

### ISSUE THREAD

**Issue by sb3nder** [validator/validator#2138]
### Summary

`<label><select><button></button></select></label>`

throws an error:
`The label element may contain at most one button, input, meter, output, progress, select, or textarea descendant.`

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<label><select><button></button></select></label>

</body>
</html>
```

### Checker error messages

```shell
Error: The label element may contain at most one button, input, meter, output, progress, select, or textarea descendant.

From line 8, column 16; to line 8, column 23

l><select><button></butt

Warning: label element with multiple labelable descendants.

From line 8, column 1; to line 8, column 7

>↩<body>↩↩<label><selec
```

### Steps to reproduce

n/a

### Expected behavior

n/a

### Actual behavior

n/a

**Comment by mtrootyy** [comment]
This error messages is correct.


The "select" element and "button" element is labelable elements.
Content model of the "label" element is "Phrasing content, but with no descendant labelable elements unless it is the element's labeled control, and no descendant label elements".
In other words, a "label" element cannot contain both a "select" element and a "button" element as descendants.



```
<label><select><button></button></select></label>
```
This "label" element labels the "select" element.
Therefore this "button" element is "descendant labelable elements unless it is the element's labeled control" of lalel element.
Therefore the markup above is unconformance.



The correct markup is as follows:
```
<label for="select1"></label>
<select id="select1"><button></button></select>
```

**Comment by sideshowbarker** [comment]
Closing per https://github.com/validator/validator/issues/2138#issuecomment-5609015748
